src/command.rs

```diff
@@ -89,37 +89,28 @@ impl<'l> YKGetHMAC<'l> {
             YKCommand::HmacSlot2 => {}
             _ => return Err(Status::IncorrectDataParameter),
         };
-        Ok(YKGetHMAC {
+        Ok(YKGetHmac {
             challenge: self.challenge,
             slot_cmd: Some(slot),
         })
     }
 }
 
-impl<'l> TryFrom<&'l [u8]> for YKGetHMAC<'l> {
+impl<'l> TryFrom<&'l [u8]> for YKGetHmac<'l> {
     type Error = Status;
     fn try_from(data: &'l [u8]) -> Result<Self, Self::Error> {
         // Input data should always be padded to 64 bytes
         ensure(data.len() == 64, Status::IncorrectDataParameter)?;
-        // PKCS#7 padding; compatibility with Yubikey's PKCS#7 version
-        // TODO extract to separate crate or use known good implementation
-        let challenge = {
-            let mut idx = data.len();
-            while idx > 0 {
-                idx -= 1;
-                // TODO OPT can use unchecked get here
-                if data[idx] != data[63] {
-                    break;
-                }
-            }
-            if idx == 0 && data[0] == data[1] {
-                // All sent is padding
-                return Err(Status::IncorrectDataParameter);
-            }
-            // We have at least one element in the challenge
-            debug_now!("Found length {}", idx);
-            &data[..=idx]
-        };
+        // PKCS#7 padding; possibly incompatible with Yubikey's PKCS#7 version, as it expects
+        // the last byte to always be the padding byte value. See KeepassXC implementation comments
```

Comment:
I think having a link here would help a lot.

Reply:
But link to what?

Reply:
https://github.com/Nitrokey/keepassxc/blob/cf819e0a3f5664fb0e1705217dbebbdf704bdc34/src/keys/drivers/YubiKeyInterfacePCSC.cpp#L730